testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/concurrent/HungTasksTerminationTestCase.java

```diff
@@ -0,0 +1,301 @@
+import org.jboss.as.ee.subsystem.ManagedExecutorTerminateHungTasksOperation;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.jboss.remoting3.security.RemotingPermission;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.naming.InitialContext;
+import java.io.FilePermission;
+import java.io.IOException;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutionException;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ALLOW_RESOURCE_SERVICE_RESTART;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILURE_DESCRIPTION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OPERATION_HEADERS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.jboss.as.test.shared.integration.ejb.security.PermissionUtils.createPermissionsXmlAsset;
+
+/**
+ * Test case for managed executor's hung tasks termination feature.
+ *
+ * @author Eduardo Martins
+ */
+@RunWith(Arquillian.class)
+public class HungTasksTerminationTestCase {
+
+    private static final Logger logger = Logger.getLogger(HungTasksTerminationTestCase.class);
+
+    private static final PathAddress EE_SUBSYSTEM_PATH_ADDRESS = PathAddress.pathAddress(PathElement.pathElement(SUBSYSTEM, EeExtension.SUBSYSTEM_NAME));
+    private static final String RESOURCE_NAME = HungTasksTerminationTestCase.class.getSimpleName();
+    // a task is considered hung when is running over 0.5s
+    private static final long HUNG_TASK_THRESHOLD_TEST_VALUE = 500;
+    // hung tasks periodic cancellation should be done every 5s
+    private static final long HUNG_TASK_CANCELLATION_PERIOD_TEST_VALUE = 5000;
+
+    @ArquillianResource
+    private ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(JavaArchive.class, HungTasksTerminationTestCase.class.getSimpleName() + ".jar")
+                .addClasses(HungTasksTerminationTestCase.class)
+                .addAsManifestResource(new StringAsset("Dependencies: org.jboss.as.controller, org.jboss.as.ee, org.jboss.remoting\n"), "MANIFEST.MF")
+                .addAsManifestResource(createPermissionsXmlAsset(
+                        new RemotingPermission("createEndpoint"),
+                        new RemotingPermission("connect"),
+                        new FilePermission(System.getProperty("jboss.inst") + "/standalone/tmp/auth/*", "read")
```

Comment:
@emmartins what is this permissions for?

Reply:
Needed for security manager runs, https://issues.redhat.com/browse/WFLY-5169 added these to the similar EEConcurrentManagementTestCase